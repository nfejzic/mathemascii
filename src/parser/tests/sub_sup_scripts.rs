use super::Snapshot;

super::test_snap!(n_to_infty, "n->oo");
super::test_snap!(n_to_infty_grouping, "(n->oo)");
super::test_snap!(complex_subscripts, "lim_(N->oo) sum_(i=0)^N");
