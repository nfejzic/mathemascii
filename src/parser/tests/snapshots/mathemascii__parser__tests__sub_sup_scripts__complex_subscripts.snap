---
source: src/parser/tests/sub_sup_scripts.rs
expression: "Snapshot((input, math))"
---
lim_(N->oo) sum_(i=0)^N

Interm {
| IntermExpr {
| | Function(Lim)
| | | sub:
| | | | OpenParen
| | | | | Interm {
| | | | | | IntermExpr {
| | | | | | | Variable("N")
| | | | | | }
| | | | | }
| | | | | 
| | | | | Interm {
| | | | | | IntermExpr {
| | | | | | | Arrow(To)
| | | | | | }
| | | | | }
| | | | | 
| | | | | Interm {
| | | | | | IntermExpr {
| | | | | | | Other(Infinity)
| | | | | | }
| | | | | }
| | | | |
| | | | CloseParen
| }
}

Interm {
| IntermExpr {
| | Operator(Sum)
| | | sub:
| | | | OpenParen
| | | | | Interm {
| | | | | | IntermExpr {
| | | | | | | Variable("i")
| | | | | | }
| | | | | }
| | | | | 
| | | | | Interm {
| | | | | | IntermExpr {
| | | | | | | Relation(Eq)
| | | | | | }
| | | | | }
| | | | | 
| | | | | Interm {
| | | | | | IntermExpr {
| | | | | | | Number("0")
| | | | | | }
| | | | | }
| | | | |
| | | | CloseParen
| | | sup:
| | | | Variable("N")
| }
}


